### DISCUSSION TITLE
[Design/General] Economy Rework [DRAFT]

### DISCUSSION THREAD

**Discussion by dearmochi** [ParadiseSS13/Paradise#17766] [Documents]
# What

Reworks the economy system to play a bigger part in the average round. These are the main ways this will express itself:
- **Almost everything should cost money**. You could get a snack at a vending machine, or get the chef to serve you something.
- There should be **more ways to earn money** besides gambling, some more reliable than others (e.g. selling things via cargo shuttle, stock market PDA minigame)
- Beyond the current uses for money (which is little and generally not game impacting), there should be **more ways to spend your hard earned cash**, that could have more impact than that we are used to.

# Why

Right now, the economy is non existent. Everyone spawns with 500 credits and that's it. By improving the system, it can be made more engaging for the players by having them _work_ for cash instead of just gambling for 30 minutes. Hard earned cash gets traded for well deserved prizes.

By giving money more importance, players would hopefully pay more attention to the system and involve money in more roleplay situations.

# How

## Ditch ATMs

> Introducing Nanotrasen's own payment system! From now on, your salary will be deposited in space-cloud vaults encrypted with only the best security (rumors of it being plain text are baseless).

~ _Cashless is the Future_

- Remove all existing ATMs and instead only have one in Cargo (the new economical hub). You can withdraw and deposit money there but it's discouraged. The ATM can also accept coins to deposit (values TBD).
- Add the NanoBank PDA app. Allows you to login via PIN or inserted ID to: view balance, transfer money, request money transfer (like EFTPOS).

## Stock Market Minigame

> Invest in promising ventures and earn enough for multiple lifetimes! Lose it all after getting greedy! Sell everything, rinse and repeat!

~ _NT-STONKS_

- Implement a Stock Market page in the bank PDA app. From there, you can invest in a few companies (5 or 6) with fixed or randomly generated names. The value of each company would fluctuate over the round and sometimes can even sharply raise/drop following fluff events that would be announced over newscasters.

## Cargo Bounties

> Our clients all over the galaxy always need something, and this is your chance to help out and make a profit! Commission rates currently at 99%.

~ _WTB Crayons_

- Players can take up a Cargo Bounty at a Cargo Bounty Board. A bounty requires you to turn over one or multiple items, mobs or structures. Once in possession, they can be brought to cargo then expedited via the Cargo Shuttle.
- Cargo employees can earn a commission (on top of the bounty taker) for handling the bounty delivery.
- Bonus money for a speedy delivery.
- Bounties should be relevant to one's job, otherwise generic bounties.

## Cargo Changes

- Remove cargo points. Instead, ordering a crate costs money and the person making the order is expected to pay for it.
- Objects or other things of value can be sold on the cargo shuttle. 
  - This can be done using a special box for selling. Upon creation the box retains the seller's bank account for deposit.
  - Cargo employees can also get a commission for handling the sale.

## Everything has a Price

- From clothes to security gear, everything should cost money. How to determine a price for each and every item is unknown still.
- If an item has a price, it most likely can be sold as well.
- Players now receive a paycheck regularly.

## Random unsorted ideas

- Inflation (creating money out of thin air is bad)
- Wad of cash becomes stronger the fatter it is
- Syndicate objective of possessing a certain amount of money
- Better sprites
- Wiki page for economy mechanics along with some fluff lore

# Implementation Details

idk yet

**Comment by Carthusia** [comment]
I very much like where this is going, is there the possibility of money carrying over between rounds?

So say every played round you get (Blank) amount, nothing to high as to not inflate the credit, and standard prices like say vending machine food and other disposable common items is priced on this standard baseline, making each credit worth a bit more, so instead of starting with our current inflated credit, where a can of soup is 50 bucks, that same can could be 5-10, a nice meal from the kitchen could run 15-20, BUT things like say, clothing from the vendors?

Well they cost more, as clothing vendors on station would more or less be Rent-a-Suits, since you dont OWN the suit as it goes away at the end of the round, theyd still be a smidge pricy, but nothing that would break the bank,  and if you return to CC with the clothing items in hand, you recieve say 75% of the cost back? 25% was the rental fee, 75% of it was the collateral.

Now, this bit would require a loadout rework to, but i think it would be a neat idea, since money carrys over each round and saves, players can buy clothing for their loadouts, this would be clothing that they OWN, and thus can spawn with it every round, clothing bought from this is much more expensive, so say a blue suit on station would run ya 100-200, buying it would be round say 1000.

The loadout system would also need all those clothing bits in the vendors added to it as well, and be priced accordingly, as right now i think only like 20-40% of the available clothing vendors items are actually in the loadout screen.

This helps money actually matter a bit and makes people want to acquire it.

There was a server i recall seeing where they had a payment system where at the end of a round you got paid, died you got say 100, if you lived you got 200, if you also completed your objectives as say an antag you got a "Criminal" bonus of like 500, and if you completed your regular civilian objective you got an extra 100-200.

**Reply by Necaladun** [reply]
Persistent economy has staggeringly huge amounts of problems with it, but the biggest is basically 'if people lose money in a round they will get salty'. Having to track this over multiple rounds and reimburse players etc would be a huge load on the admins, and also difficult to do when it comes to antags. It'd also mean that older players have advantage over newer ones which we just don't want at all.

**Reply by Carthusia** [reply]
I mean, if people loose money then that's what happens. Its entire purpose more or less in the above system would be loadout purchases with no mechanical value. 

You could place a cap on the amount they bring in at the start of the round, so say even if they have all the money in the world and could buy god, they can only bring into the round at most say 500-1k. 

The only time I could see people legitimately requesting a reimbursement would be for unscheduled manual server resets, similar to antag tokens, but even then since they only can bring in a capped amount, their total reserve is not affected, so an outright denial of reimbursement wont really anger people to much.

You can earn, steal, find more money in the round, and all that saves to your wallet, and all that can be used for loadout purchases.

Now I see this reaching a point similar to karma points, you have everything bought, and more money than you know what to do with, which is probably inevitable in any system with persistence.


Overall, persistence sounds like its a no go then.

**Comment by atteria** [comment]
There's an issue right now where new accounts can be issued with basically any amount of money, allowing the HoP to give themselves a blank check for anything. I've used this for RP things like contraband buyback programs, but if money is going to be made actually useful the account system will have to be changed to address this.

**Comment by Denthamos** [comment]
I would advise to have more than a single ATM on station at cargo just in case cargo gets blown up.  An abandoned store themed room in maintenance for example boarded up with cobwebs and such with an ATM there could work.

**Comment by KalevTait** [comment]
Currently on station there is a strong sense of co-operation. Cook gives everyone food. Hydroponics gives cook whatever he asks for. Medical heals people when they come in, no questions asked. Science helps whoever asks for help, assuming they can.

If everything cost money, all these things would go away. All food would cost money, and you'd need to either do a transaction directly with the cook, or you'd have to have a vending machine that the cook stocked with food, and set the price for. The former creates a bottleneck for getting food, while the later reduces the number of interactions on the station.

Cook meanwhile can decide whether or not he wants to buy something from hydroponics. If he does, he's just lost money, which given round length, he's not guarenteed to get back. If he doesn't hydroponics isn't making money, so can't spend as much. So we put in a vending machine hydroponics can sell into.

Pretty soon, everyone is just playing single player games that only interact in the loosest of senses.

I'm not arguing against strengthening the economy, in terms of things you can do to make money as well as what you can spend money on - but doing so at the cost of the cooperation that currently happens on station... that's not a game I'd want to play.

**Comment by Pckables** [comment]
I agree with one premise that people are saying:
Items created/supplied by the station should not cost credits (vending machine contents are imported products and thus cost money).
If people want to require cash payment for them, that has to be handled RP-wise, met with appropriate RP responses/consequences.

I like the idea that cash should be specifically for off-station items, and strongly believe that's the direction the economy idea should be expanded:
- Heavily expand what can be ordered from cargo for niche but interesting items people would want to buy, especially civilians.
- Have cargo be able to order more important/advanced departmental items for each area, but for exorbitant prices.
- Rework what science can sell for money as they're the most reliant on cargo's imports.
- Improve Cargo's Supply console UI for easier navigation, to handle the massive influx of new items.
- Have methods each department can earn "Departmental Cash" (explained below).

Departmental Cash would be sent to and handled by the Department head. If there is no department head present, it is sent to a holding account that can be accessed by the HoP/Captain. (Heads that go SSD have their cash automatically sent to the holding account)
Departments earn this cash from accomplishing tasks specific to their work (tasks to be determined).
-Science has xenobio slime types, mechs, strange objects, tech disks, etc.
-Medical might earn some from healing/fixing damage (self-inflicted doesn't count), selling injectors with only super powers active, keeping the fridge stocked healing chems, doing autopsies on crew, and developing viruses/vaccines.
-Engineering accrues it at a steady rate based on station health and power. The less breaches, and power being not too low/high means more income. Completing station objectives adds a massive bonus.
-Security accrues at a fixed rate. The captain may divert funds to security should they feel it's required.
-Cargo gets a commission from any sales, along with minerals deposited into the ORM.

- Service isn't a united department so they do not have departmental cash, rather unique way to earn personal cash.
-- Botanists are the same as now, mutated plant seeds can be sold for cash. The first of each type gives a massive bonus.
-- Bartenders accrue cash for every X minutes they spend in the bar area (not SSD). They also have a "Tip Jar" patrons can use.
-- Chefs identical to Bartender, but the kitchen rather than bar.
-- Janitors gain cash for cleaning messes/picking up trash not made by themselves (or another janitor).
-- Clowns and Mimes gain cash for every X minutes not spent in Maintenance/Security
-- Barbers, Chaplains, and Librarians accrue cash every X minutes spent in their respective room.

One important detail is to make sure the system can't be abused by repeatable actions, hence why self-harm isn't profitable and janitors can't make their own messes for money. (But if a Janitor convinces 5 people to make nonstop messes for money, that might make for a fun 'embezzlement' RP punishment).

Let me know what ideas you do/don't like. I dumped a lot more stuff there than I expected.
